pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -58,3 +75,34 @@ func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) a
 	}
 	return response
 }
+
+// getUnknownPaths returns a list of unknown paths.
+func (h *losslessDefaulter) getUnknownPaths(raw []byte) (sets.Set[string], error) {
+	obj := h.object.DeepCopyObject()
+	strictErrors, err := json.UnmarshalStrict(raw, obj, json.DisallowUnknownFields)
+	if err != nil {
+		return nil, err
+	}
+	paths := sets.New[string]()
+	for _, err := range strictErrors {
+		if fieldError, ok := err.(json.FieldError); ok {
+			paths.Insert(fieldError.FieldPath())
+		}
+	}
+	return paths, nil
+}
+
+// convertToFieldErrorPath converts the given path into the FieldError path.
+func convertToFieldErrorPath(path string) string {
+	// Replace /foo/bar to foo/bar
+	path = strings.TrimLeft(path, "/")
+	// Replace foo/0 to foo[0]
+	path = reLastArrayIndex.ReplaceAllString(path, `[$1]`)
+	// Replace foo/1/bar to foo[1].bar
+	path = reInnerArrayIndex.ReplaceAllString(path, `[$1].`)
+	// Replace foo/bar to foo.bar
+	path = strings.ReplaceAll(path, "/", ".")
+	// Replace example.com~1foo to example.com/foo
+	path = strings.ReplaceAll(path, "~1", "/")
```

Comment:
Uhm... I wonder if there are other escape characters we should be aware of. Can you check the jsonpath documentation?

Or is there a different library or setting for strict json verification that we can use?